shuup/core/models/_suppliers.py

```diff
@@ -169,23 +202,30 @@ def get_orderability_errors(self, shop_product, quantity, customer):
         :type contect: shuup.core.models.Contact
         :rtype: iterable[ValidationError]
         """
-        return self.module.get_orderability_errors(shop_product=shop_product, quantity=quantity, customer=customer)
+        return self.modules[
+            self.get_supplier_module_index_for_product_id(shop_product.product.id)
+        ].get_orderability_errors(shop_product=shop_product, quantity=quantity, customer=customer, *args, **kwargs)
 
-    def get_stock_statuses(self, product_ids):
+    def get_stock_statuses(self, product_ids, *args, **kwargs):
         """
         :param product_ids: Iterable of product IDs.
         :return: Dict of {product_id: ProductStockStatus}
         :rtype: dict[int, shuup.core.stocks.ProductStockStatus]
         """
-        return self.module.get_stock_statuses(product_ids)
+        return_dict = {}
+        for module_index, ids in self.get_supplier_module_for_product_ids(product_ids).items():
```

Comment:
remove this `get_supplier_module_for_product_ids`, the supplier should return only the stock statuses that is handled by it already:

```py
stock_statuses = dict()
for module in self.modules:
     stock_statuses.update(module.get_stock_statuses(ids, *args, **kwargs))
return stock_statuses
```